### PR TITLE
pyupgrade one-off run

### PR DESCRIPTION
**Pull request by crusaderky**
A one-off, manually vetted and tweaked run of pyupgrade

### CODE REVIEW

**Comment by crusaderky** [comment]
Ready for review and merge. Test failures are all due to codecov.io being offline.

File: xarray/backends/pseudonetcdf_.py [hunk 1]
```diff
@@ -75,17 +75,17 @@ def get_variables(self):
...
     def get_encoding(self):
         encoding = {}
-        encoding['unlimited_dims'] = set(
-            [k for k in self.ds.dimensions
-             if self.ds.dimensions[k].isunlimited()])
+        encoding['unlimited_dims'] = {
+            k for k in self.ds.dimensions
+            if self.ds.dimensions[k].isunlimited()
+        }
         return encoding
```
Comment (Zac-HD) [inline xarray/backends/pseudonetcdf_.py]: This method could use the same, more concise, version as pynio from the next file.

Comment (crusaderky) [reply xarray/backends/pseudonetcdf_.py]: Done, thanks for spotting it

**Comment by max-sixty** [comment]
Great!! Thanks @crusaderky !

**Pull request merged**